### ISSUE TITLE
File is deleted on save on exit

### ISSUE THREAD

**Issue by selsie** [Pulover/PuloversMacroCreator#179]
So I have my beautifully constructed macro with a few hundred lines (or less).
Open it again to run it.
I change something random.
Click the X to close the window.
I will be asked if I want to save it.
YES!
BOOM!!!! The Macro is gone. Wiped out without trace.
This happened a few times before, and I keep working backups, but today it hit me hard :(

This is a bug that can be reproduced. Using the newest Version 5.40

**Comment by mrodal** [comment]
YES, this happened to me too, but not like this, the program hung, which happens frequently, I went to close the process, and GOOD BYE SCRIPT... this cannot happen!!

**Comment by selsie** [comment]
> YES, this happened to me too, but not like this, the program hung, which happens frequently, I went to close the process, and GOOD BYE SCRIPT... this cannot happen!!

I keep my scripts backupped now.

**Comment by wedoprods** [comment]
Little late response but sometimes macros get corrupted.
